### PR TITLE
TC01 - User can log into Railway with valid username and password

### PR DESCRIPTION
**Pull request by thienan090196**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,5 +28,27 @@
...
+                <version>2.12.2</version>
+                <configuration>
+                    <suiteXmlFiles>
+                        <suiteXmlFile>src/test/resources/testng.xml</suiteXmlFile>
+                    </suiteXmlFiles>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 </project>
```
Comment (thanhlelgg) [inline pom.xml:54]: new line at EOF

File: src/main/java/com/logigear/page_objects/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.logigear.page_objects;
+
+import com.logigear.elements.Label;
+import org.openqa.selenium.By;
+
+public class HomePage extends BasePage {
+
+    //Elements
+    public final Label lblWelcomeMessage = new Label(By.cssSelector(".account strong"));
```
Comment (thanhlelgg) [inline src/main/java/com/logigear/page_objects/HomePage.java:9]: don't use text formatting attribute in locator unless there's no other way

File: src/main/java/com/logigear/page_objects/LoginPage.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.logigear.elements.Button;
+import com.logigear.elements.TextBox;
+import com.logigear.models.Account;
+import org.openqa.selenium.By;
+
+public class LoginPage extends BasePage {
+
+    //Elements
+    private final TextBox txtEmail = new TextBox(By.cssSelector("input#username"));
```
Comment (thanhlelgg) [inline src/main/java/com/logigear/page_objects/LoginPage.java:11]: why don't you just use By.id ?

File: src/test/java/com/logigear/tests/LoginTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Test(description = "User can log into Railway with valid username and password")
+    public void TC01() {
+        account = new Account(Constant.EMAIL, Constant.PASSWORD);
+        homePage.goToLoginPage();
+        loginPage.login(account);
+
+        String actualResult = homePage.getWelcomeMessage();
+        String expectedResult = "Welcome " + account.getEmail();
+        Assert.assertEquals(actualResult, expectedResult, actualResult + " is not matched with " + expectedResult);
```
Comment (thanhlelgg) [inline src/test/java/com/logigear/tests/LoginTest.java:24]: no need to print the compare. assertEquals will do that when the test fail

File: src/test/resources/TestNG.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+            <class name="com.logigear.tests.LoginTest"/>
+        </classes>
+    </test>
+    <test name="Firefox Test">
+        <parameter name="browser" value="firefox"/>
+        <classes>
+            <class name="com.logigear.tests.LoginTest"/>
+        </classes>
+    </test>
+</suite>
```
Comment (thanhlelgg) [inline src/test/resources/TestNG.xml:15]: new line at EOF

**Comment by thienan090196** [comment]
@thanhlelgg I updated my code based on your request. Please review and let me know if any problems

File: src/main/java/com/logigear/helpers/DriverHelper.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import java.util.concurrent.TimeUnit;
+
+public class DriverHelper {
+
+    private static final DriverHelper instance = new DriverHelper();
```
Comment (LoliSlayerMC) [inline src/main/java/com/logigear/helpers/DriverHelper.java:15]: No need to create new instance. This is a helper class, so make all the necessary methods to public static modifier.

File: src/test/java/com/logigear/tests/BaseTest.java [hunk 7]
```diff
@@ -1,15 +1,25 @@
...
+import org.testng.annotations.Parameters;
 
 public class BaseTest {
 
     @BeforeMethod
-    public void setUp() {
+    @Parameters("browser")
+    public void setUp(@Optional("chrome") String browser) {
+        DriverHelper.getInstance().createDriver(DriverType.valueOf(browser.toUpperCase()));
+        DriverHelper.setWindowSize(Constant.DEFAULT_SCREEN_WIDTH, Constant.DEFAULT_SCREEN_HEIGHT);
```
Comment (LoliSlayerMC) [inline src/test/java/com/logigear/tests/BaseTest.java:17]: We usually use `maximize()`, don't set a fixed screen size

**Review by LoliSlayerMC — CHANGES_REQUESTED** [review]

File: src/main/java/com/logigear/helpers/DriverHelper.java (hunk 6 above)
Comment (thienan090196) [reply src/main/java/com/logigear/helpers/DriverHelper.java]: @LoliSlayerMC I had refactored my code and updated the latest one. Let me know if you have any questions or comments.

File: src/test/java/com/logigear/tests/BaseTest.java (hunk 7 above)
Comment (thienan090196) [reply src/test/java/com/logigear/tests/BaseTest.java]: @LoliSlayerMC I had replaced the set window size to maximize. Please review and feedback to me

**Review by LoliSlayerMC — APPROVED** [review]
LGTM

**Pull request merged**